src/Adapter/CrudTrait.php

```diff
@@ -68,22 +74,44 @@ public function update($table, array $data, $where = '', array $bind = [])
      * Delete from table.
      *
      * @param string $table
-     * @param string $where
-     * @param array $bind
+     * @param array|string $where Deprecated usage of parameter as string
+     * @param array $bind Deprecated in favour of $where as array
      * @return int Number of affected rows
      */
-    public function delete($table, $where = '', array $bind = [])
+    public function delete($table, $where = [], array $bind = [])
     {
         $table = $this->quote()->identifier($table);
-        $sql   = "DELETE FROM $table"
-            . (($where) ? " WHERE $where" : '');
+        $sql   = "DELETE FROM $table";
+        if (is_array($where)) {
+            $where = $this->createWhereExpression($where);
+        }
+        $sql .= (($where) ? " WHERE {$where}" : '');
 
         $stmt = $this->query($sql, $bind);
 
         return $stmt->rowCount();
     }
 
     /**
+     * Create WHERE expression from given criteria
+     *
+     * @param array $where
+     * @return string
+     */
+    private function createWhereExpression(array $where = [])
+    {
+        $criteria = [];
+        foreach ($where as $str => $value) {
+            if (is_int($str)) {
+                $criteria[] = $value;
+            } else {
+                $criteria[] = $this->quote()->into($str, $value);
+            }
+        }
+        return implode(' AND ', $criteria);
```

Comment:
Can't quite decide if this is hard coded and rigid or acceptable give the use case.

Reply:
Yeah, I thought the same. But I think it's acceptable for the simple operations this is aimed at.

Theoretically still possible to so more complex criteria, passing a single string (with values already quoted, etc) in the *where* array.